src/HDFFile.cpp

```diff
@@ -39,63 +39,104 @@ static void writeAttribute(hdf5::node::Node &Node, const std::string &Name,
 }
 
 template <typename DT>
-static void appendValue(const nlohmann::json *Value, std::vector<DT> &Buffer) {
-  if (Value->is_number_integer()) {
-    Buffer.push_back(Value->get<int64_t>());
-  } else if (Value->is_number_unsigned()) {
-    Buffer.push_back(Value->get<uint64_t>());
-  } else if (Value->is_number_float()) {
-    Buffer.push_back(Value->get<double>());
+static void appendValue(nlohmann::json const &Value, std::vector<DT> &Buffer) {
+  if (Value.is_number_integer()) {
+    Buffer.push_back(Value.get<int64_t>());
+  } else if (Value.is_number_unsigned()) {
+    Buffer.push_back(Value.get<uint64_t>());
+  } else if (Value.is_number_float()) {
+    Buffer.push_back(Value.get<double>());
   } else {
-    std::throw_with_nested(std::runtime_error(fmt::format(
-        "Expect a numeric value but got: {}", Value->dump().substr(0, 256))));
-  }
-}
-
-template <typename DT>
-static std::vector<DT> populateBlob(const nlohmann::json *Value,
-                                    hssize_t GoalSize) {
-  std::vector<DT> Buffer;
-  if (Value->is_array()) {
-    std::stack<json const *> as;
-    std::stack<size_t> ai;
-    std::stack<size_t> an;
-    as.push(Value);
-    ai.push(0);
-    an.push(Value->size());
-
-    while (!as.empty()) {
-      if (as.size() > 10) {
+    auto What = fmt::format("Expect a numeric value but got: {}",
+                            Value.dump().substr(0, 256));
+    std::throw_with_nested(std::runtime_error(What));
+  }
+}
+
+class StackItem {
+public:
+  StackItem(nlohmann::json const &Value) : Value(Value), Size(Value.size()) {}
+  void inc() { ++Index; }
+  nlohmann::json const &value() { return Value.at(Index); }
+  bool exhausted() { return !(Index < Size); }
+
+private:
+  nlohmann::json const &Value;
+  size_t Index = 0;
+  size_t Size = 0;
+};
+
+static size_t const MAX_DIMENSIONS_OF_ARRAY = 10;
+
+template <typename _DataType> class NumericItemHandler {
+public:
+  using DataType = _DataType;
+  static void append(std::vector<DataType> &Buffer, nlohmann::json const &Value,
+                     size_t const ItemLength = 0) {
+    appendValue(Value.get<DataType>(), Buffer);
+  }
+};
+
+class StringItemHandler {
+public:
+  using DataType = std::string;
+  static void append(std::vector<DataType> &Buffer, nlohmann::json const &Value,
+                     size_t const ItemLength = 0) {
+    Buffer.push_back(Value);
+  }
+};
+
+class FixedStringItemHandler {
+public:
+  using DataType = char;
+  static void append(std::vector<DataType> &Buffer, nlohmann::json const &Value,
+                     size_t const ItemLength = 0) {
+    if (ItemLength >= 1024 * 1024) {
```

Comment:
Can you define this max length somewhere outside the function? Maybe put `MAX_DIMENSIONS_OF_ARRAY` and this together at the beginning of the file makes "less hidden" these magic numbers